x-pack/plugins/lists/server/services/exception_lists/exception_list_client.test.ts

```diff
@@ -32,4 +43,108 @@ describe('exception_list_client', () => {
       expect(listItem).toEqual(getExceptionListItemSchemaMock());
     });
   });
+
+  describe('server extension points execution', () => {
+    let extensionPointStorageContext: ExtensionPointStorageContextMock;
+    let exceptionListClient: ExceptionListClient;
+
+    beforeEach(() => {
+      extensionPointStorageContext = createExtensionPointStorageMock();
+    });
+
+    // Test client methods that use the `pipeRun()` method of the ExtensionPointStorageClient`
+    describe.each([
+      [
+        'createExceptionListItem',
+        (): ReturnType<ExceptionListClient['createExceptionListItem']> => {
+          return exceptionListClient.createExceptionListItem(
+            getCreateExceptionListItemOptionsMock()
+          );
+        },
+        (): ExtensionPointStorageContextMock['exceptionPreCreate']['callback'] => {
+          return extensionPointStorageContext.exceptionPreCreate.callback;
+        },
+      ],
+
+      [
+        'updateExceptionListItem',
+        (): ReturnType<ExceptionListClient['updateExceptionListItem']> => {
+          return exceptionListClient.updateExceptionListItem(
+            getUpdateExceptionListItemOptionsMock()
+          );
+        },
+        (): ExtensionPointStorageContextMock['exceptionPreUpdate']['callback'] => {
+          return extensionPointStorageContext.exceptionPreUpdate.callback;
+        },
+      ],
+    ])(
+      'and calling `ExceptionListClient#%s()`',
+      (methodName, callExceptionListClientMethod, getExtensionPointCallback) => {
+        describe('and server extension points are enabled', () => {
+          beforeEach(() => {
+            exceptionListClient = new ExceptionListClient({
+              disableServerExtensionPoints: false,
+              savedObjectsClient: getExceptionListSavedObjectClientMock(),
+              serverExtensionsClient:
+                extensionPointStorageContext.extensionPointStorage.getClient(),
+              user: 'elastic',
+            });
+          });
+
+          it('should execute extension point callbacks', async () => {
+            await callExceptionListClientMethod();
+
+            expect(getExtensionPointCallback()).toHaveBeenCalled();
+          });
+
+          it('should validate extension point callback returned data and throw if not valid', async () => {
+            const extensionPointCallback = getExtensionPointCallback();
+            // @ts-expect-error
```

Comment:
I have not been able to find a solution around this one. I have tried multiple ways of csting both input and output with no success. Any help would be appreciated 🙏 

![image](https://user-images.githubusercontent.com/56442535/148119499-18f40772-5f6a-4327-baef-9c8854cc77f0.png)

Reply:
I don't 100% follow the logic but this should work out:

```ts
            extensionPointCallback.mockImplementation(async (args) => {
              // exclude entries
              const { entries, ...rest } = args as CreateExceptionListItemOptions &
                UpdateExceptionListItemOptions;
              expect(entries).toBeTruthy(); // Test entries to exist since we exclude it.
              return rest as CreateExceptionListItemOptions & UpdateExceptionListItemOptions;
            });
```

Reply:
😲 
Nice. this works. Thanks for the help 🙏